packages/excalidraw/actions/actionDuplicateSelection.tsx

```diff
@@ -254,41 +258,30 @@ const duplicateElements = (
     oldElements,
     oldIdToDuplicatedId,
   );
+  // elementsWithClones
+  //   .filter((ele) =>
+  //     oldElements.some((e) => oldIdToDuplicatedId.get(e.id) === ele.id),
+  //   )
+  //   .forEach((ele) => {
+  //     mutateElement(ele, { strokeColor: "rgba(8, 255, 0, 1)" });
+  //   });
+
   fixBindingsAfterDuplication(
-    elementsWithClones,
+    finalElements,
```

Comment:
Both finalElements and elementsWithClones are exactly the same; there’s no difference between them. No matter which one you use, it’s going to work the same. I chose finalElements because it’s the array returned from the function at the end. However, finalElements is derived from elementsWithClones. I’m not sure why the original author chose to use elementsWithClones, especially since the array is reversed here so that the duplicated element is rendered above the original one,

if you want the duplicate one to be behind the original then you dont need to reverse and you can just return the elementsWithClone array,

so basically we dont need to rewrite everything in fixBindingAfterDuplication

``
  const finalElementsReversed: ExcalidrawElement[] = [];

  const finalElementIds = new Map<ExcalidrawElement["id"], true>();
  index = elementsWithClones.length;

  while (--index >= 0) {
    const element = elementsWithClones[index];
    if (!finalElementIds.get(element.id)) {
      finalElementIds.set(element.id, true);
      finalElementsReversed.push(element);
    }
  }

  // step (3)
  const finalElements = syncMovedIndices(
    finalElementsReversed.reverse(),
    arrayToMap(newElements),
  );
``

``
  return {
    elements: finalElements,
    appState: {
      ...appState,
    },
  };
``

Reply:
i have made some  new changes like,
all some mutateFunction because i wanstnt seeing a good result in different condition